src/InferOpt.jl

```diff
@@ -14,6 +14,7 @@ using SparseArrays
 using Statistics
 using StatsBase: StatsBase, sample
 using Test
+using ThreadsX
```

Comment:
Julia 1.8 lets you check how much time package loading takes, have you tried it to see if ThreadsX changes anything?

Reply:
Do you have a link on how to do that ?

Reply:
https://docs.julialang.org/en/v1/NEWS/#InteractiveUtils

Reply:
If I correctly interpreted the log, `ThreadsX` adds up only 20ms of compile time:

```julia
julia> @time_imports using InferOpt
[ Info: Precompiling InferOpt [4846b161-c94e-4150-8dac-c7ae193c601f]
      0.2 ms  Compat
     63.8 ms  ChainRulesCore
     20.1 ms  GenericSchur
      2.9 ms  ProgressMeter
      0.2 ms  ExprTools
     37.0 ms  TimerOutputs
     17.1 ms  MacroTools
      1.7 ms  StaticArraysCore
      0.4 ms  ConstructionBase
     20.3 ms  Setfield
    226.0 ms  MutableArithmetics
     60.7 ms  Parsers 8.15% compilation time
     11.6 ms  JSON
      1.8 ms  BenchmarkTools
      3.2 ms  IrrationalConstants
      0.4 ms  DiffRules
      1.4 ms  DiffResults
    394.3 ms  StaticArrays
      0.1 ms  OpenLibm_jll
      0.2 ms  NaNMath
      2.0 ms  DocStringExtensions 52.06% compilation time
      0.5 ms  ChangesOfVariables
      0.8 ms  InverseFunctions
      0.5 ms  LogExpFunctions
      0.2 ms  CompilerSupportLibraries_jll
      0.3 ms  OpenSpecFun_jll
     11.0 ms  SpecialFunctions
      0.2 ms  CommonSubexpressions
     80.6 ms  ForwardDiff
     49.2 ms  DataStructures
      1.3 ms  TranscodingStreams
      0.3 ms  Bzip2_jll
      0.5 ms  CodecBzip2
      0.1 ms  Zlib_jll
      0.8 ms  CodecZlib
   1285.7 ms  MathOptInterface
      0.1 ms  Hungarian
      2.2 ms  OpenBLAS_jll 94.37% compilation time
      0.3 ms  Arpack_jll
      5.6 ms  Arpack
    580.6 ms  FrankWolfe
      9.2 ms  Krylov
      0.1 ms  FastClosures
     22.5 ms  AMD 93.99% compilation time
      4.0 ms  LDLFactorizations
     28.0 ms  LinearOperators
      1.2 ms  SimpleTraits
      0.9 ms  DataAPI
      0.3 ms  SortingAlgorithms
      7.5 ms  Missings
      0.2 ms  StatsAPI
     19.2 ms  StatsBase
      8.2 ms  SplittablesBase
      0.4 ms  ArgCheck
     23.9 ms  InitialValues
      0.2 ms  Requires
      0.1 ms  DataValueInterfaces
      0.1 ms  IteratorInterfaceExtensions
      0.0 ms  TableTraits
     16.2 ms  Tables
      0.2 ms  ZygoteRules
    529.1 ms  BangBang 95.37% compilation time (71% recompilation)
      8.1 ms  MicroCollections
      0.2 ms  Adapt
      2.2 ms  Referenceables
     59.1 ms  Baselet
      0.1 ms  CompositionsBase
      0.1 ms  DefineSingletons
     54.4 ms  Transducers 42.80% compilation time
     18.6 ms  ThreadsX
      3.9 ms  InferOpt

```

Reply:
Amazing!